semgrep-core/Engine/Semgrep.ml

```diff
@@ -130,11 +135,34 @@ let (mini_rule_of_pattern: R.t -> (R.pattern_id * Pattern.t) -> MR.t) =
     languages =
       (match r.R.languages with
        | R.L (x, xs) -> x::xs
-       | R.LNone | R.LGeneric -> failwith "TODO: LNone | LGeneric"
+       | R.LNone | R.LGeneric -> raise Impossible
```

Comment:
For this, I use `assert false`. One advantage is that the exception contains the source location, even if no stack trace is collected or printed. Another advantage is that it's standard, see https://caml.inria.fr/pub/docs/manual-ocaml/expr.html#sss:expr-assertion